quill-sql/src/main/scala/io/getquill/source/sql/idiom/SqlIdiom.scala

```diff
@@ -162,8 +162,10 @@ trait SqlIdiom {
   implicit def orderByCriteriaShow(implicit strategy: NamingStrategy): Show[OrderByCriteria] = new Show[OrderByCriteria] {
     def show(criteria: OrderByCriteria) =
       criteria match {
-        case OrderByCriteria(prop, true)  => s"${prop.show} DESC"
-        case OrderByCriteria(prop, false) => prop.show
+        case OrderByCriteria(prop, AscNullsFirst)  => s"${prop.show} ASC NULLS FIRST"
```

Comment:
The sql generation always defines explicit nulls ordering as it can vary according to the database implementation.